python/src/equistore/operations/equal_metadata.py

```diff
@@ -240,6 +240,9 @@ def _check_blocks(a: TensorBlock, b: TensorBlock, props: List[str], fname: str):
                 if not (c1.names == c2.names):
                     raise ValueError(err_msg + err_msg_names)
 
+                if not (len(c1) == len(c2)):
+                    raise ValueError(err_msg + err_msg_1)
+
```

Comment:
I'm not sure I see the point of this check. Doesn't the check on line 246 do the same thing?
```py
if not np.all(c1 == c2):
    raise ValueError(err_msg + err_msg_1)
```

Reply:
The line below does check the same thing, but creates a warning about elementwise comparison in numpy.